packages/cli/build/typescript.ts

```diff
@@ -0,0 +1,34 @@
+import { runCli } from '@magic-works/i18n-codegen'
+import { spawn, SpawnOptions } from 'child_process'
+import { resolve } from 'path'
+import { ROOT_PATH } from '../utils'
+import makeLockSequence from './process-lock'
+
+const config = resolve(ROOT_PATH, '.i18n-codegen.json')
+const options: SpawnOptions = {
+    cwd: ROOT_PATH,
+    stdio: 'inherit',
+    shell: true,
+}
+
+export async function dev() {
+    lock: for await (const lock of makeLockSequence()) {
+        if (await lock()) break lock
+    }
+    runCli({ config, watch: true }, console.error)
+    // start a one-time "tsc -b" or parallel "tsc -b -w"
+    return spawn('npx', ['tsc', '-b', '-w'], options)
```

Comment:
TypeScript might die. Would you like to add code to restart tsc when it dies?

Reply:
yourself solve fix this problem?

Reply:
well not urgent.